services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -455,6 +482,49 @@ impl SentinelS2L2aCogsMetaData {
         }
         debug!("number of generated loading infos: {}", parts.len());
 
+        // This is a workaround to avoid errors when a provider does not fill the complete query rectangle.
+        let buffered_start =
+            query.time_interval.start() + Duration::seconds(self.stac_query_buffer.start_seconds);
```

Comment:
I would make the start_seconds positive and subtract here.

Reply:
changed it (sorry no extra commit)